### PR TITLE
Renamed BusExtensions to MessageHandlerContextExtensions

### PR DESCRIPTION
**Pull request by danielmarbach**
@Particular/nservicebus-maintainers please review and merge

### CODE REVIEW

**Comment by timbussmann** [comment]
any value in obsoleting `BusExtensions`?

**Comment by danielmarbach** [comment]
@timbussmann I pushed an update. It made me realize that we never properly obsoleted the `BusExtension` methods before. We just changed the signature :( I will run a doco update

File: src/NServiceBus.Gateway/obsoletes.cs [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
+}
 
 namespace NServiceBus.Gateway.Routing
 {
     [ObsoleteEx(
        Message = "Not used anymore. Use config.Gateway().ChannelFactories() to provide custom channel factories if you want to override the gatways default http implementation.",
-       RemoveInVersion = "7.0",
-       TreatAsErrorFromVersion = "6.0")]
+       RemoveInVersion = "3.0",
+       TreatAsErrorFromVersion = "2.0")]
```
Comment (timbussmann) [inline src/NServiceBus.Gateway/obsoletes.cs]: good catch :+1:

**Comment by timbussmann** [comment]
LGMT :shipit:

**Pull request merged**